### ISSUE TITLE
Detection with Torch Hub Failing

### ISSUE THREAD

**Issue by llavkush** [ultralytics/yolov5#13519]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I have ML project on Python 3.7.8, it was working fine, but recently I am getting error when loading model with  torch.hub.load 

 File "C:\Users\Strange/.cache\torch\hub\ultralytics_yolov5_master\models\common.py", line 39, in <module>
    from utils.dataloaders import exif_transpose, letterbox
  File "C:\Users\Strange/.cache\torch\hub\ultralytics_yolov5_master\utils\dataloaders.py", line 776
    if mosaic := self.mosaic and random.random() < hyp["mosaic"]:

I know yolov5 require python > 3.8, but all current dependencies in my project is of 3.7.8, please suggest workaround to work with it

**Comment by UltralyticsAssistant** [comment]
👋 Hello @llavkush, thank you for your interest in YOLOv5 🚀! It looks like you're encountering an issue while using YOLOv5 with Torch Hub. As you noted, YOLOv5 indeed requires Python >= 3.8 to function properly, and supporting Python 3.7.8 is outside the recommended environment.

If this is a 🐛 Bug Report, please share a **minimum reproducible example (MRE)** of your code that demonstrates the issue. This will greatly help us to investigate further.

In the meantime, here are some helpful resources and recommendations:

## Requirements

YOLOv5 requires [**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) dependencies installed:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

If you're unable to upgrade your Python version, consider running YOLOv5 in one of our verified environments where all dependencies are preinstalled:

## Environments

YOLOv5 can be run in the following up-to-date verified environments:
- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Using one of these environments may allow you to bypass the need to modify your local setup.

## Tutorials

For additional guidance:
- ⭐️ [YOLOv5 Tutorials](https://docs.ultralytics.com/yolov5/)
- 🤖 [Torch Hub Integration](https://pytorch.org/hub/ultralytics_yolov5/)

Let us know if upgrading your environment or utilizing a verified environment resolves your issue. If not, please provide the requested MRE for further debugging. This is an automated response – an Ultralytics engineer will take a closer look and assist you further soon. 😊

**Comment by pderrenger** [comment]
@llavkush to resolve the Python 3.7 compatibility issue with YOLOv5's walrus operator (`:=`), we recommend either:

1. **Upgrading to Python ≥3.8** (officially required for YOLOv5), or  
2. **Using YOLOv5 v6.2** which maintained Python 3.7 compatibility:  
```python
model = torch.hub.load('ultralytics/yolov5:v6.2', 'yolov5s')
```

For details see the YOLOv5 v6.2 release notes. We strongly recommend updating to Python 3.8+ for continued support with modern versions.

**Comment by llavkush** [comment]
@pderrenger thank you for the solution, this works.

**Comment by pderrenger** [comment]
@llavkush you're welcome! Glad the solution worked for you. For future reference, our YOLOv5 documentation contains detailed version compatibility notes and migration guides if you decide to upgrade Python versions later. Happy coding! 🚀
